cmd/compute/instance/instance_reset.go

```diff
@@ -65,15 +70,27 @@ func (c *instanceResetCmd) CmdRun(_ *cobra.Command, _ []string) error {
 		}
 	}
 
-	opts := make([]egoscale.ResetInstanceOpt, 0)
+	request := v3.ResetInstanceRequest{}
 
 	if c.DiskSize > 0 {
-		opts = append(opts, egoscale.ResetInstanceWithDiskSize(c.DiskSize))
+		request.DiskSize = c.DiskSize
 	}
 
-	var template *egoscale.Template
 	if c.Template != "" {
-		template, err = globalstate.EgoscaleClient.FindTemplate(ctx, c.Zone, c.Template, c.TemplateVisibility)
+
+		var templates *v3.ListTemplatesResponse
+
+		if c.TemplateVisibility != "" {
+			templates, err = client.ListTemplates(ctx, v3.ListTemplatesWithVisibility(v3.ListTemplatesVisibility(c.TemplateVisibility)))
+		} else {
+			templates, err = client.ListTemplates(ctx)
+		}
```

Comment:
We specify default value in command description:
```
instance template visibility (public|private) (default "public")
```

Being explicit in code as well works a bit better IMO:
```go
if c.TemplateVisibility == "" {
    c.TemplateVisibility = "public"
}

templates, err := client.ListTemplates(ctx, v3.ListTemplatesWithVisibility(v3.ListTemplatesVisibility(c.TemplateVisibility)))
```